docs/examples.rst

```diff
@@ -103,8 +103,154 @@ specification is obeyed, meaning that you can do things like provide a
+        'sync_plan': <nailgun.entity_fields.OneToOneField object at 0x7fb5bd49eac8>,
+    }
+
+Fields serve two purposes. First, they provide typing information mixins. For
+example, a server expects this JSON payload when creating a product::
+
+    {
+        "name": "junk product",
+        "organization_id": 5,
+        …
+    }
+
+And a server will return this JSON payload when reading a product::
+
+
+    {
+        "name": "junk product",
+        "organization": {
+            'id': 3,
+            'label': 'c5f2646f-5975-48c4-b2a3-bf8398b44510',
+            'name': 'junk org',
+        },
+        …
+    }
+
+Notice how the "organization" field is named and structured differently in the
+above two cases. NailGun can deal with this irregularity due to the presence of
+the ``StringField`` and ``OneToOneField``. If you are ever fiddling with an
+entity's definition, be careful to use the right field types. Otherwise, you may
+get some strange and hard-to-troubleshoot bugs.
+
+Secondly, fields can generate random values for unit testing purposes. (This
+does *not* normally happen!) See the ``create_missing`` method for more
+information.
+
+``read``
+~~~~~~~~
+
+Rather unsurprisingly, the ``read`` method fetches information about an entity.
+Importantly, the ``read`` method does not have side-effects:
+
+    >>> from nailgun import entities
+    >>> org = entities.Organization(id=418)
+    >>> response = org.read()
+    >>> for obj in (org, response):
+    ...     type(obj)
+    ...
+    <class 'nailgun.entities.Organization'>
+    <class 'nailgun.entities.Organization'>
+    >>> for obj in (org, response):
+    ...     obj.get_values()
+    ...
+    {'id': 418}
+    {
+        'description': None,
+        'id': 418,
+        'label': 'junk_org',
+        'name': 'junk org',
+        'title': 'junk org',
+    }
+
+As demonstrated above, the ``read`` method does not alter the object it is
+called on. Instead, it creates a new object, populates that object with
+attributes, and returns that new object. As a result, idioms like ``org =
+org.read()`` are advisable.
+
+``update``
+~~~~~~~~~~
+
+The ``update`` method updates an entity's values. For example:
+
+    >>> from nailgun.entities import Organization
+    >>> org = Organization(id=418).read()
+    >>> org.get_values()
+    {
+        'description': None,
+        'id': 418,
+        'label': 'junk_org',
+        'name': 'junk org',
+        'title': 'junk org',
+    }
+    >>> org.name = 'junkier org'
+    >>> org.description = 'supercalifragilisticexpialidocious'
+    >>> org = org.update()  # update all fields by default
+    >>> org.get_values()
+    {
+        'description': 'supercalifragilisticexpialidocious',
+        'id': 418,
+        'label': 'junk_org',
+        'name': 'junkier org',
+        'title': 'junkier org',
+    }
+    >>> org.description = None
+    >>> org = org.update(['description'])  # update only named fields
```

Comment:
I think that a method which requires an iterable of field names is [easier to comprehend](http://docs.python-guide.org/en/latest/writing/style/#explicit-code) and will lead to generally nicer client code. What do I mean by "nicer client code?" Consider the case where I want to update all of an organization's fields except it's name. You could do so like this:

```python
org = …  # org is an Organization object.
org = org.update(set(org.get_values().keys()) - set(['name']))
```

If `*args` were required, the same code would look like this:

```python
org = …  # org is an Organization object.
org = org.update(*(set(org.get_values().keys()) - set(['name'])))
```